pkg/hintrunner/zero/zerohint_dictionaries_test.go

```diff
@@ -37,5 +37,37 @@ func TestZeroHintDictionaries(t *testing.T) {
 				errCheck: errorTextContains("assertion `len(keys) == 0` failed"),
 			},
 		},
+		"SquashDictInnerNextKey": {
+			{
+				operanders: []*hintOperander{
+					{Name: "next_key", Kind: uninitialized},
+				},
+				ctxInit: func(ctx *hinter.HintRunnerContext) {
+					err := ctx.ScopeManager.AssignVariable("keys", []fp.Element{})
+					if err != nil {
+						t.Fatal(err)
+					}
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newSquashDictInnerNextKeyHint(ctx.operanders["next_key"])
+				},
+				errCheck: errorTextContains("no keys left but remaining_accesses > 0"),
+			},
+			{
+				operanders: []*hintOperander{
+					{Name: "next_key", Kind: uninitialized},
+				},
+				ctxInit: func(ctx *hinter.HintRunnerContext) {
+					err := ctx.ScopeManager.AssignVariable("keys", []fp.Element{*feltUint64(1), *feltUint64(2), *feltUint64(3)})
+					if err != nil {
+						t.Fatal(err)
+					}
+				},
+				makeHinter: func(ctx *hintTestContext) hinter.Hinter {
+					return newSquashDictInnerNextKeyHint(ctx.operanders["next_key"])
+				},
+				check: varValueEquals("next_key", feltUint64(3)),
```

Comment:
we have to check for the value of keys in the scope, so I thinks its a good idea to add new utility

Reply:
yes thats what i supposed! thanks